dpctl/_sycl_device.pyx

```diff
@@ -286,7 +286,11 @@ cdef class SyclDevice(_SyclDevice):
     def print_device_info(self):
         """ Print information about the SYCL device.
         """
-        DPCTLDeviceMgr_PrintDeviceInfo(self._device_ref)
+        cdef const char * info_str = DPCTLDeviceMgr_GetDeviceInfoStr(
+            self._device_ref
+        )
+        py_info = <bytes> info_str
```

Comment:
info_str needs to be freed.

Reply:
Do we have a function in DPCTL to do that? It was allocated using `new`, so I can not delete it using `PyMem_free`.

Reply:
https://github.com/IntelPython/dpctl/blob/782340f798db864c89b5247753af4c421d78a30c/dpctl-capi/include/dpctl_utils.h#L40

Reply:
Fixed.